prow/gerrit/client/syncer.go

```diff
@@ -146,25 +150,78 @@ func (st *SyncTime) Update(newState LastSyncState) error {
 		}
 	}
 
-	if !changed {
-		return nil
+	st.val = targetState
+
+	// Pediodically sync time back to storage instead of writing after every
+	// single project is processed, as the I/O would become performance
+	// bottleneck when there are lots of projects.
+	if st.stateChan == nil {
+		st.stateChan = make(chan LastSyncState)
+		go func() {
```

Comment:
Concurrency is hard for me to understand, so give me some rope here. But AFAICS we do not call `wg.Wait()` on this goroutine, so for every SyncTime receiver we will create a new goroutine. Do these ever get garbage-collected? That is, is this a possible source of leaking goroutines?

Reply:
there is a single goroutine for each client, added a comment